### PR TITLE
add GHA workflows to replace Drone

### PR DESCRIPTION
**Pull request by jiaqiluo**
This PR adds the GHA workflows to replace the existing Drone CI, and also updates the existing dispatch workflow to meet the new requirements. 

For validation, a modified version of the code is used to build and push on the event of tag in my fork:
- CI run: https://github.com/jiaqiluo/system-agent-installer-k3s/actions/runs/9423600375
- the tag in DockerHub: https://hub.docker.com/layers/jq20/system-agent-installer-k3s/v1.29.5-k3s1/images/sha256-a71315ed3b9ce644e1f657f17f0fc3468a9b8ed2ef596456f146d088f351d774?context=repo

I don't have an easy way to validate the workflow `Create Tag on Dispatch`.  I will keep monitoring it after the PR is merged.

### CODE REVIEW

**Review by dereknola — APPROVED** [review]

File: scripts/download [hunk 1]
```diff
@@ -26,6 +26,7 @@ if [[ -z "${LOCAL_ARTIFACTS}" ]]; then
         SUFFIX="-${ARCH}"
     fi
     curl -sfL -R -o artifacts/k3s https://github.com/k3s-io/k3s/releases/download/${URI_VERSION}/k3s${SUFFIX}
+    ls artifacts
```
Comment (slickwarren) [inline scripts/download:29]: is this needed, or just added for debugging?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      id-token: write # needed for the Vault authentication
+    needs: [ build-linux ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set environment variables
+        run: |
+          . scripts/version
+          echo "VERSION=$VERSION"         >> "$GITHUB_ENV"
```
Comment (slickwarren) [inline .github/workflows/release.yaml:80]: this is in the scripts/version file too, I think one could be removed?

File: scripts/download (hunk 1 above)
Comment (jiaqiluo) [reply scripts/download]: yep. It was removed.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (jiaqiluo) [reply .github/workflows/release.yaml]: removed

**Review by slickwarren — APPROVED** [review]

**Review by brandond — CHANGES_REQUESTED** [review]
LGTM, but requesting changes to make sure this is blocked until after the current release is done.

**Review by brandond — APPROVED** [review]

**Pull request merged**